src/promptflow/promptflow/_telemetry/activity.py

```diff
@@ -51,19 +52,19 @@ def log_activity(
     :type custom_dimensions: dict
     :return: None
     """
-    from promptflow._core.operation_context import OperationContext
-
-    user_agent = OperationContext.get_instance().get_user_agent()
-
+    custom_dimensions = custom_dimensions or {}
+    context = OperationContext.get_instance()
+    request_id = custom_dimensions.get('request_id', context.get('request_id', str(uuid.uuid4())))
+    user_agent = context.get_user_agent()
     activity_info = {
         # TODO(2699383): use same request id with service caller
-        "request_id": str(uuid.uuid4()),
+        "request_id": request_id,
         "activity_name": activity_name,
         "activity_type": activity_type,
-        "user_agent": user_agent,
+        "user_agent": user_agent
     }
-    custom_dimensions = custom_dimensions or {}
     activity_info.update(custom_dimensions)
+    context['request_id'] = request_id
```

Comment:
If you want to store `request_id` in context, maybe you can set it when get None.

Reply:
Yes, if custom_dimensions has request_id, I'll set it to context.
Priority:
`request id of custom_dimensions` >` request id of context `>` create a new request id.`
And then the latest request id will be set to context.

Reply:
I mean, should we only set request id to context when there's no request id in context? there's no need to store request id if we won't use.

Reply:
Because the logs we type have a request id, using context to save the request id helps all logs use the same request id, which is beneficial for us to query logs.

Reply:
I think in our sync, we decied that 
we can have a operation id(session id) that stays the same for a python process. 
But request_id shall change for each request send to server?

Reply:
We have already discussed it offline. Here, we will use the` request id` first, because in the entire function call chain, it is possible that some functions may not pass the `request id` in their `custom_dimensions `(for example, other developers may not pass the `request id` when developing new functions). This way, we can save the "request ID" from the previous `custom_dimensions `and use it, It is possible to ensure that the same `request id` is used throughout the entire function call chain, which is beneficial for querying logs

After adding the `operation id` later, this field can be added so that the `operation id` can be used to query the logs in the entire process, and then the `request id` can be used to distinguish the logs in different function call chains